src/strategy/periodic.rs

```diff
@@ -41,15 +58,57 @@ impl StrategyPeriodic {
             n => log::trace!("periodic updates, weekly calendar length: {} minutes", n),
         };
 
-        let strategy = Self { schedule: calendar };
+        let strategy = Self {
+            schedule: calendar,
+            time_zone: tz,
+        };
         Ok(strategy)
     }
 
+    /// Get the time zone from `periodic` strategy config.
+    fn get_time_zone(cfg: &inputs::PeriodicInput) -> Fallible<Tz> {
+        let tz;
+        if &cfg.time_zone == "localtime" {
+            // Use `read_link()` instead of `exists()` because we only want to check for
+            // the existence of the `/etc/localtime` symlink, not whether it points to
+            // a valid file. (`read_link()` returns an error if symlink doesn't exist)
+            if read_link(Path::new("/etc/localtime")).is_err() {
+                tz = Tz::named("UTC");
+            } else {
+                tz = Tz::local();
+            }
+        } else {
+            tz = Tz::named(&cfg.time_zone)
+        }
+        tz.map_err(|err| err.into())
+    }
+
     /// Return the measured length of the schedule, in minutes.
     pub(crate) fn schedule_length_minutes(&self) -> u64 {
         self.schedule.length_minutes()
     }
 
+    /// Return the weekday and time of the next window, in human terms.
+    pub(crate) fn human_next_window(&self) -> String {
+        let naive_utc_dt = Utc::now().naive_utc();
+        let dt = (&self.time_zone).from_utc_datetime(&naive_utc_dt);
+        let next_window_minute_in_week = self.schedule.next_window_minute_in_week(&dt);
+
+        match next_window_minute_in_week {
+            Some(minute_in_week) => {
+                let (weekday, hour, minute) = utils::weekly_minute_as_weekday_time(minute_in_week);
+                // `tmp_dt` is created just to use the `%Z` specifier of its `format()` method.
+                let tmp_dt = (&self.time_zone).from_utc_datetime(&Utc::now().naive_utc());
+                let tz_abbrev = tmp_dt.format("%Z").to_string();
+                format!(
+                    "at {}:{} on {} ({}) if no shift in clock.",
```

Comment:
Up for your consideration, but you may turn the last part into a `(subject to timezone caveats)` if not-UTC.

Reply:
That's a good idea, since the next reboot window information isn't _always_ wrong even in the week of a shift in clock.